source/features/review-comments-hidden-indicator.tsx

```diff
@@ -0,0 +1,131 @@
+import './review-comments-hidden-indicator.css';
+import React from 'dom-chef';
+import select from 'select-dom';
+import features from '../libs/features';
+import observeEl from '../libs/simplified-element-observer';
+import {comment as commentIcon} from '../libs/icons';
+
+const SELECTOR_COMMENT_CONTAINER = 'tr.inline-comments';
+const SELECTOR_CUSTOM_TOGGLE = 'tr.refined-toggle-comments';
+const SELECTOR_COMMENT = '.review-comment .js-comment';
+
+let isVisible = true; // Are comments visible?
+
+const getOffsetY = (el: Element): number =>
+	el.getBoundingClientRect().top + window.pageYOffset;
+
+// Toggle comments while maintaining scroll position
+const toggleComments = (target: HTMLElement): void => {
+	const container = target.closest(SELECTOR_CUSTOM_TOGGLE)!;
+	const scrollOld = window.scrollY;
+	const offsetOld = getOffsetY(container);
+
+	select.all<HTMLInputElement>('input.js-toggle-file-notes').forEach(input => {
```

Comment:
`toggleComments` looks for _all_ the comments. This shouldn't happen. If you click one of the indicators, only the comments on that file should be restored.

Also there's a utility function called `anchorScroll` that you can use to avoid the scroll. Example usage:

```js
const showCommentsHandler = (event): void => {
	anchorScroll(() => {
		(event.target as HTMLElement)
			.closest(SELECTOR_CUSTOM_TOGGLE)!
			.nextElementSibling!
			.querySelector('.js-toggle-file-notes')!
			.click();
	});
};
```

⬆️ This is probably what the whole function will look like.

Reply:
> toggleComments looks for all the comments. This shouldn't happen. If you click one of the indicators, only the comments on that file should be restored.

toggleComments actually looks for all files (not comments) using `input.js-toggle-file-notes` (default GitHub checkbox to toggle comments) and toggles using it only.

> If you click one of the indicators, only the comments on that file should be restored.

Sure we want that behaviour? I would've preferred hiding all comments. Fine with what you say.

> Also there's a utility function called `anchorScroll`

Thanks. Will look into it.

Reply:
Yes, that’s what I meant, all the comments on the page will be restored and that’s not necessary. Only the comments on the current file should (because that’s how the default “Show comments” works